build.gradle

```diff
@@ -107,23 +127,43 @@ List<String> resolveIntelliJVersions(String versionKey) {
   return [versionValue]
 }
 
-class ResolveIntelliJVersions extends DefaultTask {
-  @Input
-  @Option(option = 'version-key', description = 'Either release number (like 2020.3) or key of intellijVersions (like eapOfLatestSupportedMajor)')
-  String versionKey
-
-  @OptionValues('version-key')
-  List<String> getAvailableVersionKeys() {
-    return new ArrayList<>(project.intellijVersions.keySet())
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Element
+import org.jsoup.nodes.Document
+
+String checkForNewerEap(String intellijMajorVersion) {
+  Document htmlContent = Jsoup.connect(project.intellijSnapshotsUrl).get()
+  Element link = htmlContent.select('a[href^='+project.intellijSnapshotsUrl+'com/jetbrains/intellij/idea/ideaIC/'+intellijMajorVersion+'.][href$="-EAP-SNAPSHOT.pom"]')[0]
+  if (link) {
+    String attr = link.attr("href")
+    Pattern pattern = Pattern.compile("ideaIC-(.*?).pom")
+    Matcher matcher = pattern.matcher(attr)
+    matcher.find()
+    return matcher.group(1)
+  } else {
+    return null
   }
+}
 
-  @TaskAction
-  void print() {
-    println(String.join("\n", project.resolveIntelliJVersions(versionKey)))
-  }
+String newEapVersion() {
+  char[] latestStable = project.intellijVersions.latestStable.toCharArray()
+  return Character.getNumericValue(latestStable[5]) == 3 ? Character.toString(latestStable[2]) + Character.toString(latestStable[3] + 1) + "1":
```

Comment:
Pls note that in general we should not rely on there being **3** major releases with the given year (although it's been a practice so far)... hence I suggested filtering the available EAP versions for one that is **higher** than the provided one! Rather than looking for a specific hand-crafted version via a regex, as done now 🤔